chapter04/rpi/ex2.pl

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+sub check_items_for_all {
+  my ( $all ) = @_;
+
+  foreach $name ( keys %$all ) {
```

Comment:
In newer versions of perl ``keys`` will accept a hashref, so there's no need to dereference here most of the time.